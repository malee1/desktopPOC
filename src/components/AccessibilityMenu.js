import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import { Box, Typography, Button } from '@material-ui/core';
import { setAccessibility } from '../redux/actions';

const useStyles = makeStyles({
  root: {
    backgroundColor: 'white',
    width: '50vw',
    height: '40vh'
  },
  title: { padding: '15px' },
  options: { padding: '20px' },
  buttonContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%'
  },
  button: {
    width: '80%',
    fontSize: '40px'
  }
});

export default function AccessibilityMenu() {
  const classes = useStyles();

  const dispatch = useDispatch();
  const optionOneSelection = useSelector((state) => state.accessibility.accessibility);

  const handleOptionOneClick = () => {
    dispatch(setAccessibility(!optionOneSelection));
  };

  return (
    <div className={classes.root} data-testid="accessibility-menu">
      <Box>
        <Typography className={classes.title} variant="h4">
          Accessibility Menu
        </Typography>
        <Typography className={classes.options} variant="h6">
          Please select a display option
        </Typography>
        <Box className={classes.buttonContainer}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            onClick={handleOptionOneClick}
            data-testid="button"
          >
            Toggle Display Option One
          </Button>
        </Box>
      </Box>
    </div>
  );
}
