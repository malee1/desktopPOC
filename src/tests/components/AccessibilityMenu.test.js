import React from 'react';

import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';

import AccessibilityMenu from '../../components/AccessibilityMenu';

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('panel renders correctly', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<AccessibilityMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has one component', () => {
    const { getAllByTestId } = render(<AccessibilityMenu />);
    const container = getAllByTestId('accessibility-menu');
    expect(container.length).toBe(1);
  });
  it('has one button', () => {
    const { getAllByTestId } = render(<AccessibilityMenu />);
    const container = getAllByTestId('button');
    expect(container.length).toBe(1);
  });
});
