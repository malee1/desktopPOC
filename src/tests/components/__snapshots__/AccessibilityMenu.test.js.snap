// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel renders correctly matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="accessibility-menu"
>
  <div
    className="MuiBox-root MuiBox-root-6"
  >
    <h4
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h4"
    >
      Accessibility Menu
    </h4>
    <h6
      className="MuiTypography-root makeStyles-options-3 MuiTypography-h6"
    >
      Please select a display option
    </h6>
    <div
      className="MuiBox-root MuiBox-root-7 makeStyles-buttonContainer-4"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Display Option One
        </span>
      </button>
    </div>
  </div>
</div>
`;
